lib/src/json_schema/constants.dart

```diff
@@ -158,23 +158,37 @@ class SchemaVersion implements Comparable<SchemaVersion> {
   }
 }
 
-Map getJsonSchemaDefinitionByRef(String ref) {
-  final mapping = {
-    SchemaVersion.draft4.toString(): JsonSchemaDefinitions.draft4,
-    SchemaVersion.draft6.toString(): JsonSchemaDefinitions.draft6,
-    SchemaVersion.draft7.toString(): JsonSchemaDefinitions.draft7,
-    SchemaVersion.draft2019_09.toString(): JsonSchemaDefinitions.draft2019_09,
-  };
+final _staticSchemaMapping = {
+  Uri.parse(SchemaVersion.draft4.toString()).removeFragment(): JsonSchemaDefinitions.draft4,
+  Uri.parse(SchemaVersion.draft6.toString()).removeFragment(): JsonSchemaDefinitions.draft6,
+  Uri.parse(SchemaVersion.draft7.toString()).removeFragment(): JsonSchemaDefinitions.draft7,
+  Uri.parse(SchemaVersion.draft2019_09.toString()).removeFragment(): JsonSchemaDefinitions.draft2019_09,
+  Uri.parse("https://json-schema.org/draft/2019-09/meta/validation"): Draft2019Subschemas.validation,
+  Uri.parse("https://json-schema.org/draft/2019-09/vocab/validation"): Draft2019Subschemas.validation,
+  Uri.parse("https://json-schema.org/draft/2019-09/meta/format"): Draft2019Subschemas.format,
+  Uri.parse("https://json-schema.org/draft/2019-09/vocab/format"): Draft2019Subschemas.format,
+  Uri.parse("https://json-schema.org/draft/2019-09/meta/core"): Draft2019Subschemas.core,
+  Uri.parse("https://json-schema.org/draft/2019-09/vocab/core"): Draft2019Subschemas.core,
+  Uri.parse("https://json-schema.org/draft/2019-09/meta/metadata"): Draft2019Subschemas.metadata,
+  Uri.parse("https://json-schema.org/draft/2019-09/vocab/metadata"): Draft2019Subschemas.metadata,
+  Uri.parse("https://json-schema.org/draft/2019-09/meta/applicator"): Draft2019Subschemas.applicator,
+  Uri.parse("https://json-schema.org/draft/2019-09/vocab/applicator"): Draft2019Subschemas.applicator,
+  Uri.parse("https://json-schema.org/draft/2019-09/meta/content"): Draft2019Subschemas.content,
+  Uri.parse("https://json-schema.org/draft/2019-09/vocab/content"): Draft2019Subschemas.content,
+};
 
-  if (SchemaVersion.values.map((value) => value.toString()).contains(ref)) {
-    return json.decode(mapping[ref]);
-  }
+Map getStaticSchema(String ref) {
+  return getStaticSchemaByURI(Uri.parse(ref));
+}
 
-  return null;
+Map getStaticSchemaByURI(Uri ref) {
+  if (ref.fragment != "") return null;
```

Comment:
wonder if we should handle the http / https versions of each?

Reply:
Good call!